Jetsnack/app/src/main/java/com/example/jetsnack/ui/home/Cart.kt

```diff
@@ -16,20 +16,323 @@
+import com.example.jetsnack.ui.utils.statusBarsHeight
 
 @Composable
-fun Cart(modifier: Modifier = Modifier) {
-    Text(
-        text = stringResource(R.string.home_cart),
+fun Cart(
+    onSnackClick: (Long) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val cartSnacks = remember { SnackRepo.getCart() }
+    val inspiredByCart = remember { SnackRepo.getInspiredByCart() }
+    Cart(cartSnacks, inspiredByCart, onSnackClick, modifier)
+}
+
+@Composable
+fun Cart(
+    cartSnacks: List<Snack>,
+    inspiredByCart: SnackCollection,
+    onSnackClick: (Long) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    JetsnackSurface(modifier = modifier.fillMaxSize()) {
+        Box {
+            CartContent(cartSnacks, inspiredByCart, onSnackClick)
+            DestinationBar()
+            CheckoutBar(Modifier.align(Alignment.BottomCenter))
+        }
+    }
+}
+
+@OptIn(ExperimentalLazyDsl::class)
+@Composable
+private fun CartContent(
+    cartSnacks: List<Snack>,
+    inspiredByCart: SnackCollection,
+    onSnackClick: (Long) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    LazyColumn(modifier) {
+        item {
+            Spacer(Modifier.statusBarsHeight(additional = 56.dp))
+            Text(
+                text = stringResource(R.string.cart_order_header, cartSnacks.size),
+                style = MaterialTheme.typography.h6,
+                color = JetsnackTheme.colors.brand,
+                maxLines = 1,
+                overflow = TextOverflow.Ellipsis,
+                modifier = Modifier
+                    .preferredHeightIn(min = 56.dp)
+                    .padding(horizontal = 24.dp, vertical = 4.dp)
+                    .wrapContentHeight()
+            )
+        }
+        items(cartSnacks) { cartSnack ->
+            CartItem(cartSnack, onSnackClick)
+        }
+        item {
+            SummaryItem()
+        }
+        item {
+            SnackCollection(
+                snackCollection = inspiredByCart,
+                onSnackClick = onSnackClick,
+                highlight = false
+            )
+            Spacer(Modifier.height(56.dp))
+        }
+    }
+}
+
+@Composable
+fun CartItem(
+    snack: Snack,
+    onSnackClick: (Long) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val (count, updateCount) = remember { mutableStateOf(1) }
+    ConstraintLayout(
         modifier = modifier
-            .fillMaxSize()
-            .wrapContentSize()
-    )
+            .fillMaxWidth()
+            .clickable { onSnackClick(snack.id) }
+            .padding(horizontal = 24.dp)
+    ) {
+        val (divider, image, name, tag, priceSpacer, price, remove, quantity) = createRefs()
+        createVerticalChain(name, tag, priceSpacer, price, chainStyle = ChainStyle.Packed)
+        SnackImage(
+            imageUrl = snack.imageUrl,
+            modifier = Modifier
+                .preferredSize(100.dp)
+                .constrainAs(image) {
+                    linkTo(
```

Comment:
Nit: this might look better using the `top.linkTo(parent.top, margin = 16.dp)` syntax